src/lib.rs

```diff
@@ -36,6 +36,23 @@ pub trait KeyBox: Index {
     fn verify(&self, msg: &[u8], sgn: &Self::Signature, index: NodeIndex) -> bool;
 }
 
+pub trait MultiKeychain: KeyBox {
+    type PartialMultisignature: Debug + Clone + Encode + Decode;
+    fn get_signers(partial: &Self::PartialMultisignature) -> BoolNodeMap;
+    fn add_signature(
+        partial: &mut Self::PartialMultisignature,
+        index: NodeIndex,
+        signature: &Self::Signature,
+    );
+    fn is_complete(partial: &Self::PartialMultisignature) -> bool;
```

Comment:
It seems that `get_signers`, `add_signature`, and `is_complete` could form a trait `PartialMultisignature` as those methods don't depend on `MultiKeychain`. Is the current version more convenient for some reason?

Reply:
I wanted to avoid having an extra trait, but probably the extra trait would actually improve the readability

Reply:
On the side note, I can't see a scenario, where we use a PartialSignature without the set of node indices who have signed it, so it seems natural that the list of the signers is a part of the PartialSignature. Probably it would be more consistent to require that the Signature contains the index of the signer? This would simplify the interface in all places where we pass both the index and the signature. The only problem is a minor redundancy: when we sign a FullUnit, it already contains the index, but if we really don't like this redundancy, but maybe it would be better to remove it and only store it in the signature?

Reply:
1. In threshold signatures we don't need to track the list of signers
2. I like the idea of putting index into signature and changing full unit so that it doesn't carry the index.

Reply:
Ad 1. true, but I think we already gave up the threshold signatures.
Ad 2. Awesome, I will do it that way then.

Reply:
In what sense we gave up the threshold signatures? You mean to fit in this trait? We'll definitely use them for randomness.

Reply:
Oh, I didn't get that part. @timorl kept sayng that the threshold signatures require too complicated set up, and we want to avoid using them. If we want to use them, then probably we don't want the `get_signers` method.